includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -857,7 +858,17 @@ private function is_class_allowed_in_amp_date_picker( $class ) {
 	public function init( $sanitizers ) {
 		parent::init( $sanitizers );
 
-		foreach ( $sanitizers as $sanitizer ) {
+		$this->sanitizers = $sanitizers;
+	}
+
+	/**
+	 * Sanitize CSS styles within the HTML contained in this instance's Dom\Document.
+	 *
+	 * @since 0.4
+	 */
+	public function sanitize() {
+		// Capture the selector conversion mappings from the other sanitizers.
+		foreach ( $this->sanitizers as $sanitizer ) {
```

Comment:
Any reason for moving this foreach loop from `init()` to `sanitize()`?

Reply:
Yes, because it allows us to gather up the selector mappings _after_ all of the previous sanitizers have run, as opposed to gathering them _before_ the other sanitizers have run. This is important for example if the script sanitizer enables `native_img_used` for the `AMP_Img_Sanitizer`, because when enabled no longer does the style sanitizer need to remap `img` to `amp-img`/`amp-anim`.